scripts/keys.coffee

```diff
@@ -10,95 +10,333 @@
+		else 
+			msg.send "Dear sir/mam, You dont even have a key,you Donkey"
 
-
-	robot.respond /(.+) (has|have) (the key|key|keys|a key)/i , (msg)->
+    #this section belongs to "$name has color keys"
+	robot.respond /(.+) (has the|have the|has|have) (.+) (key|keys)/i , (msg)->
 		othername = msg.match[1]
+		keyname = msg.match[3]
 		name = msg.message.user.name
-		k = key()
 		unless othername in ["who", "who all","Who", "Who all" , "i" , "I" , "i don't" , "i dont" , "i do not" , "I don't" , "I dont" , "I do not"]
-			if othername is 'you'
-				msg.send "How am I supposed to take those keys? #{name} is a liar!"
-			else if othername is robot.name
-				msg.send "How am I supposed to take those keys? #{name} is a liar!"
-			else
-				users = robot.brain.usersForFuzzyName othername
-				if users.length is 1
-					robot.brain.set "key", users[0].name
-					msg.send "Okay, so now the keys are with #{users[0].name}"
-				else if users.length > 1
-					msg.send getAmbiguousUserText users
+			unless keyname not in ["red", "blue", "yellow", "green", "orange"]
+				users = robot.brain.userForName othername
+				if users is null
+					key_holder = null
 				else
-					msg.send "#{othername}? Never heard of 'em"
-
-	robot.respond /(i|I) (have given|gave|had given) (the key|key|keys|a key|the keys) to (.+)/i , (msg)->
+					key_holder = users.name
+				if key_holder is null
+					if othername is 'you'
+						msg.send "How am I supposed to take those #{keyname} keys? #{name} is a liar!"
+					else
+						msg.send "I don't know anyone by the name #{othername}"
+				else if othername is robot.name
+						msg.send "That's utter lies! How can you blame a bot to have the keys?"
+				else
+					if keyname is 'red'
+						robot.brain.set("keyRed",key_holder)
+					else if keyname is 'blue'
+						robot.brain.set("keyBlue",key_holder)
+					else if keyname is 'yellow'
+						robot.brain.set("keyYellow",key_holder) 
+					else if keyname is 'green'
+						robot.brain.set("keyGreen",key_holder) 
+					else if keyname is 'orange'
+						robot.brain.set("keyOrange",key_holder) 
+					msg.send "Okay, so now the #{keyname} keys are with #{othername}"	
+    #this section belongs to "i gave the keys to $name"
+	robot.respond /(i|I) (have given the|had given the|have given|gave the|had given|gave) (key|keys) to (.+)/i , (msg)->
 		othername = msg.match[4]
 		name = msg.message.user.name
-		k = key()
+		key_holder1 = keyRed()
+		key_holder2 = keyBlue()
+		key_holder3 = keyYellow()
+		key_holder4 = keyGreen() 
+		key_holder5 = keyOrange()
+		keyname=""
+		users = robot.brain.userForName othername
+		if name is key_holder1 and users.name is not "bro"
+			if users is null
+				key_holder1 = null
+				robot.brain.set("keyRed",key_holder1)
+				keyname="red"
+				msg.send "I don't know anyone by the name #{othername}"
+			else
+				key_holder1 = users.name
+				robot.brain.set("keyRed",key_holder1)
+				keyname="red"
+		else if name is key_holder2 and users.name is not "bro"
```

Comment:
`users.name is not "bro"` instead of using this every time make a parent if-else statement which checks for this condition